libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/Coders.hs

```diff
@@ -130,6 +132,22 @@ data Field t where
 field :: (x -> t -> t) -> Decode ('Closed d) x -> Field t
 field update dec = Field update (decode dec)
 
+{-# INLINE fieldGuarded #-}
+fieldGuarded ::
+  -- | The message to use if the condition fails
+  String ->
+  -- | The condition to guard against
+  (x -> Bool) ->
+  (x -> t -> t) ->
+  Decode ('Closed d) x ->
+  Field t
+fieldGuarded failMsg check update dec =
+  Field
+    update
+    ( decode dec >>= \x ->
+        x <$ unless (check x) (fail failMsg)
```

Comment:
I made this suggestion already `unless` is wrong here. We need to fail when the field is empty, not the other way around
```suggestion
        x <$ when (check x) (fail failMsg)
```

Reply:
Ah, yes! I got it backwards :+1:  :bow: